File: Core/YABACore/src/commonMain/kotlin/dev/subfly/yabacore/state/creation/notemark/NotemarkMentionCreationUIState.kt
package dev.subfly.yabacore.state.creation.notemark

import androidx.compose.runtime.Immutable
import dev.subfly.yabacore.model.ui.BookmarkUiModel

@Immutable
data class NotemarkMentionCreationUIState(
    val mentionText: String = "",
    val selectedBookmarkId: String? = null,
    val selectedBookmark: BookmarkUiModel? = null,
    val isEdit: Boolean = false,
    val editPos: Int? = null,
)

File: Darwin/YABACore/State/Creation/Notemark/NotemarkMentionCreationUIState.swift
//
//  NotemarkMentionCreationUIState.swift
//  YABACore
//

import Foundation

public struct NotemarkMentionCreationUIState: Sendable {
    public var mentionText: String
    public var selectedBookmarkId: String?
    /// Resolved from `@Query` or passed from selection; mirrors Compose `BookmarkUiModel` for display.
    public var selectedBookmarkLabel: String?
    public var isEdit: Bool
    public var editPos: Int?

    public init(
        mentionText: String = "",
        selectedBookmarkId: String? = nil,
        selectedBookmarkLabel: String? = nil,
        isEdit: Bool = false,
        editPos: Int? = nil
    ) {
        self.mentionText = mentionText
        self.selectedBookmarkId = selectedBookmarkId
        self.selectedBookmarkLabel = selectedBookmarkLabel
        self.isEdit = isEdit
        self.editPos = editPos
    }
}
